bokeh/client/websocket.py

```diff
@@ -44,17 +47,15 @@
 class WebSocketClientConnectionWrapper:
     ''' Used for compatibility across Tornado versions and to add write_lock'''
 
-    def __init__(self, socket):
-        if socket is None:
-            raise ValueError("socket must not be None")
```

Comment:
For the most part yes, especially in public user facing APIs, but here I don't see a need for one. First, because this is internal API and won't be used interactively (e.g. in IPython), so we can rely on the type system to tell us if we are using it incorrectly. Second, if this is supposed to be fully useful, then and `isinstance()` check would be my preferred way to proceed, but given the first point, I would consider this redundant.

Reply:
In general, we will need to establish a pattern for which APIs we rely solely on the type system and were we need both. However, there are libraries that can use static type annotations and generate run-time type checks, so we may want to consider those at some point, to reduce redundancy and a possibility of mismatch between those two types of type checking.